components/proxy/src/main/java/com/hotels/styx/routing/handlers/RouteRefLookup.java

```diff
@@ -0,0 +1,50 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.routing.handlers;
+
+import com.hotels.styx.api.Eventual;
+import com.hotels.styx.api.HttpHandler;
+import com.hotels.styx.routing.RoutingObjectRecord;
+import com.hotels.styx.routing.config.RoutingObjectReference;
+import com.hotels.styx.routing.db.StyxObjectStore;
+
+import java.util.function.Function;
+
+import static com.hotels.styx.api.HttpResponse.response;
+import static com.hotels.styx.api.HttpResponseStatus.NOT_FOUND;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Resolves a routing object reference from route database.
+ */
+public class RouteRefLookup implements Function<RoutingObjectReference, HttpHandler> {
```

Comment:
I would consider creating a interface for this instead of implementing Function directly. This is not as much to have a slightly easier to read code but basically a solution to overcome the problem that the IDEs 'Find usages' will find any Function instead of only Function<RoutingObjectReference, HttpHandler>